container-engine-lib/lib/database_accessors/enclave_db/file_artifacts_db/file_artifacts_db.go

```diff
@@ -0,0 +1,166 @@
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteFullUuidIndex(shortenedUuid string, targetArtifactIdx int) {
+	artifactUuids := fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid]
+	fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid] = append(artifactUuids[0:targetArtifactIdx], artifactUuids[targetArtifactIdx+1:]...)
+}
+
+func (fileArtifactDb *FileArtifactPersisted) SetContentMd5(artifactName string, md5 []byte) {
+	fileArtifactDb.data.ArtifactContentMd5[artifactName] = md5
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetContentMd5(artifactName string) ([]byte, bool) {
+	value, found := fileArtifactDb.data.ArtifactContentMd5[artifactName]
+	return value, found
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetContentMd5Map() map[string][]byte {
+	return fileArtifactDb.data.ArtifactContentMd5
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteContentMd5(artifactName string) {
+	delete(fileArtifactDb.data.ArtifactContentMd5, artifactName)
+}
+
+func GetOrCreateNewFileArtifactsDb() (*FileArtifactPersisted, error) {
+	data := fileArtifactData{
+		map[string]string{},
+		map[string][]string{},
+		map[string][]byte{},
+	}
+	db, err := enclave_db.GetOrCreateEnclaveDatabase()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get enclave database")
+	}
+	fileArtifactPersisted, err := GetFileArtifactsDbFromEnclaveDb(db, &data)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to hydrate pre-existing file artifacts")
+	}
+	return fileArtifactPersisted, nil
+}
+
+func GetFileArtifactsDbFromEnclaveDb(db *enclave_db.EnclaveDB, data *fileArtifactData) (*FileArtifactPersisted, error) {
+	err := db.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucket(takenIpAddressBucketName)
+		if err != nil && err != bolt.ErrBucketExists {
+			return stacktrace.Propagate(err, "An error occurred while creating file artifact bucket")
+		}
+		if err != bolt.ErrBucketExists {
+			if err = bucket.Put(simpleKey, consts.EmptyValueForJsonSet); err != nil {
+				return stacktrace.Propagate(err, "An error occurred while creating updating artifact bucket o '%v'", consts.EmptyValueForJsonSet)
+			}
+			return nil
+		}
+		content := tx.Bucket(takenIpAddressBucketName).Get(simpleKey)
+		if err := json.Unmarshal(content, data); err != nil {
+			return stacktrace.Propagate(err, "An error occurred restoring previous file artifact db state from '%v'", content)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating file artifact db")
+	}
+	// Bucket does exist, skipping population step
+	if err == bolt.ErrBucketExists {
+		logrus.Debugf("Taken IP addresses loaded from database")
+	} else {
+		logrus.Debugf("Taken IP addresses saved to database")
+	}
+	return &FileArtifactPersisted{
+		db,
+		data,
+	}, nil
+}
+
+func GetFileArtifactsDbForTesting(db *enclave_db.EnclaveDB, nameToUuid map[string]string) (*FileArtifactPersisted, error) {
```

Comment:
I think this method could be private

Reply:
Unfortunately this is used on unit tests :/